alacritty_terminal/src/term/mod.rs

```diff
@@ -1574,18 +1572,15 @@ impl<T: EventListener> Handler for Term<T> {
         let num_cells = (self.cols() - destination).0;
 
         let line = cursor.point.line;
-        let row = &mut self.grid[line];
-
-        unsafe {
-            let src = row[source..].as_ptr();
-            let dst = row[destination..].as_mut_ptr();
+        let line = &mut self.grid[line];
 
-            ptr::copy(src, dst, num_cells);
+        for offset in (0..num_cells).rev() {
+            line[destination + offset] = line[source + offset].clone();
```

Comment:
Same thing here, this just seems unnecessarily inefficient even for safe Rust. Cloning cells is just extremely wasteful when there's no reason why we would ever need to clone it.

Reply:
We can't really swap them because of potentially overlapping regions, the only way we can use swap here is if we do that only for non overlapping region that will be cleared later on, and after we've passed overlapping region we should go back to clone, since we literally cloning after that.